File: src/main/java/com/orange/craftsmanship/alarming/domain/DeviceRepositoryPort.java
package com.orange.craftsmanship.alarming.domain;

import java.util.Optional;

public interface DeviceRepositoryPort {
    /**
     * Get information for one device.
     *
     * @param deviceName the device name we are looking for
     * @return the alarm information, if any
     */
    Optional<Device> findDevice(String deviceName);

    /**
     * Updates devices's alerts count.
     *
     * @param deviceName  the device to be updated
     * @param alertsCount the number of alerts impacting this device
     */
    void updateAlertsCount(String deviceName, int alertsCount);

    /**
     * Updates devices's most failing neighbour.
     *
     * @param deviceName the device to be updated
     * @param neighbour  the most failing neighbour device
     */
    void updateMostFailingNeighbour(String deviceName, Device neighbour);
}

File: device_repository_port.go
package main

type Device struct {
    name                 string
    alertsCount          int
    mostFailingNeighbour *Device
}

type deviceRepositoryPort interface {
    /**
     * Get information for one Device.
     *
     * @param deviceName the Device name we are looking for
     * @return the alarm information, if any, or an empty Device struct (how to deal with optional results properly?? Using nil??)
     */
    findDevice(deviceName string) Device

    /**
     * Updates devices's alerts count.
     *
     * @param deviceName  the Device to be updated
     * @param alertsCount the number of alerts impacting this Device
     */
    updateAlertsCount(deviceName string, alertsCount int)

    /**
     * Updates devices's most failing neighbour.
     *
     * @param deviceName the Device to be updated
     * @param neighbour  the most failing neighbour Device
     */
    updateMostFailingNeighbour(deviceName string, neighbour Device)
}
